regexp/regexp.go

```diff
@@ -0,0 +1,243 @@
+package regexp
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+)
+
+// Module exposes the regexp module
+var Module = &starlarkstruct.Module{
+	Name: "regexp",
+	Members: starlark.StringDict{
+		"compile": starlark.NewBuiltin("compile", compile),
+	},
+}
+
+// The regular expression used to indentify all the backreferences in a replacement pattern.
+var backreferenceRe = regexp.MustCompile(`((\\\\)*)\\(\d)`)
+
+func compile(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		pattern string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &pattern); err != nil {
+		return nil, err
+	}
+
+	re, err := regexp.Compile(pattern)
+	if err != nil {
+		return nil, err
+	}
+	return &Regexp{re: re}, nil
+}
+
+func toList(strs []string) *starlark.List {
+	elems := make([]starlark.Value, len(strs))
+	for i, s := range strs {
+		elems[i] = starlark.String(s)
+	}
+	return starlark.NewList(elems)
+}
+
+// A Regexp represents a compiled RE2 regular expression.
+type Regexp struct {
+	re *regexp.Regexp
+}
+
+// String implements the Stringer interface.
+func (r *Regexp) String() string { return r.re.String() }
+
+// Type returns a short string describing the value's type.
+func (r *Regexp) Type() string { return "regexp" }
+
+// Freeze renders r immutable. Required by starlark.Value interface.
+// The interface regex presents to the Starlark runtime renders it immutable,
+// making this a no-op.
+func (r *Regexp) Freeze() {}
+
+// Hash returns a function of x such that Equals(x, y) => Hash(x) == Hash(y)
+// required by starlark.Value interface.
+func (r *Regexp) Hash() (uint32, error) { return starlark.String(r.re.String()).Hash() }
+
+// Truth always returns true for a Regexp.
+func (r *Regexp) Truth() starlark.Bool { return true }
+
+// Attr gets a value for a string attribute, implementing dot expression support
+// in Starklark. required by starlark.HasAttrs interface.
+func (r *Regexp) Attr(name string) (starlark.Value, error) {
+	return builtinAttr(r, name, regexMethods)
+}
+
+// AttrNames lists available dot expression strings for time. Required by
+// starlark.HasAttrs interface.
+func (r *Regexp) AttrNames() []string { return builtinAttrNames(regexMethods) }
+
+var regexMethods = map[string]*starlark.Builtin{
+	"find":            starlark.NewBuiltin("find", find),
```

Comment:
A big question we keep coming back to: how much do we want to be just like python? The more our library can be a drop-in replacement for python, the easier it will be for users to get comfortable with starlark. In particular, I always use the `match` method in python, as most docs and tutorials reference it early on. Should we be implementing that as well?

Reply:
Yeah, we should have this chat before merging a breaking change like this. Let's leave this PR in limbo for the moment